### ISSUE TITLE
Bug: High CPU usage caused by ProtonMail (and possibly other services?)

### ISSUE THREAD

**Issue by juliag2** [ferdium/ferdium-app#131]
### Avoid duplicates

- [X] I have searched the [issues tracker](https://github.com/ferdium/ferdium-app/issues) for a bug report similar to mine, in vain

### Ferdium Version

6.0.0.nightly.31

### What Operating System are you using?

Other Linux

### Operating System Version

5.17.5-arch1

### What arch are you using?

x64

### Expected Behavior

Ferdium on wayland utilizes as much CPU as chromium on wayland with a few tabs open.

### Actual Behavior

When I open ferdium and add a few services, after a short while my fan springs into action and when I check htop, it reveals that ferdium utilizes an entire thread, for reasons unknown to me.

### Steps to reproduce

It should be noted that I run ferdium with the `--enable-features=UseOzonePlatform --ozone-platform=wayland --enable-features-WaylandWindowDecorations` options. The bug does not appear to affect non-wayland instances. (Edit: It does, it just doesn't do so as fast)

1. Run `ferdium`
2. Open `htop`
3. Sort by maximum CPU utilization

### Debug link

https://debug.ferdium.org/735bad7f-3a58-46dd-b297-b9adc4f5df6f

**Comment by enx1998** [comment]
Hi all,
I have the same issue with gnome 41 wayland and ferdium 6.0.0.nightly.33

**Comment by LManMusic** [comment]
Can confirm the same issue on Windows 11 64 bit
6.0.0-nightly.39

Tried disabling the services one by one but it didn't make a difference. Services in use: WhatsApp, Messenger, Discord, Slack

**Comment by juliag2** [comment]
The issue appears to be resolved for me (6.0.0-nightly.43)

**Comment by LManMusic** [comment]
Confirm the issue is gone in nightly 43.

**Comment by juliag2** [comment]
Welp, the issue seems to be back, but only when I'm logged into the wifi of my nursing school, apparently. Or maybe my laptop just doesn't like it there.

**Comment by enx1998** [comment]
With nightly 43 I have the same problem.

**Comment by etrnal70** [comment]
I got somewhat a workaround for this. After starting Ferdium, check htop and choose the Ferdium process with the highest cpu usage, then kill it with SIGTERM.

It doesn't affect Ferdium so far, and the app still can run normally. Might be some bug in electron. I'm using sway and running it with the corresponding args

**Comment by Zeit08** [comment]
Same bug here, Ferdium.rpm 6.0.0 beta1 on Fedora 36 Gnome 42 (AMD CPU)

**Comment by juliag2** [comment]
> I got somewhat a workaround for this. After starting Ferdium, check htop and choose the Ferdium process with the highest cpu usage, then kill it with SIGTERM.
> 
> It doesn't affect Ferdium so far, and the app still can run normally. Might be some bug in electron. I'm using sway and running it with the corresponding args

I did that, and I found out that the process utilizing so much of my CPU is actually just the ProtonMail service, as the ProtonMail service tab crashes when I kill the process... I'd love to know why :D

**Comment by etrnal70** [comment]
Ah yes it's protonmail. I also have that service but didn't realize until now that it's the culprit lol. Might be good to update the issue title i guess

**Comment by juliag2** [comment]
> Ah yes it's protonmail. I also have that service but didn't realize until now that it's the culprit lol. Might be good to update the issue title i guess

Oh wow didn't even know you could do that :D

**Comment by Nitrousoxide** [comment]
Had this issue as well.   Deleted the proton service and CPU usage dropped to sane levels.  Something definitely up with its handling of that.

**Comment by pepper3k** [comment]
I can confirm as well that with only protonmail and protoncalendar services running, task manager was showing 2 ferdium.exe processes running my i7 at 13% each. As soon as I disabled them, CPU usage went back down to idle.

Windows 7 x64, latest updates.

6.0.0-nightly.54

**Comment by pepper3k** [comment]
I can also confirm that after restarting all services the problem returned. After disabling both protonmail and protoncalendar, the issue was resolved.

I opened protonmail in Chromium (iridium) and at first I saw an iridium.exe process at 13% even well after the page had finished loading. Eventually after about a minute or two, the process went down to 0%.

I know that Proton made some big changes this past month, so it could be an issue on their end. 

I think it's worth investigating and reaching out to proton.

**Comment by palepinkdot** [comment]
> I can also confirm that after restarting all services the problem returned. After disabling both protonmail and protoncalendar, the issue was resolved.
> 
> I opened protonmail in Chromium (iridium) and at first I saw an iridium.exe process at 13% even well after the page had finished loading. Eventually after about a minute or two, the process went down to 0%.
> 
> I know that Proton made some big changes this past month, so it could be an issue on their end.
> 
> I think it's worth investigating and reaching out to proton.

This happened pre-rebrand from Proton. I stopped using Ferdi / Ferdium for Proton awhile back because of this issue. I think I mentioned it to the dev team... now it seems like it's not isolated based on #219

**Comment by palepinkdot** [comment]
After taking a bit of a dive into it it looks like it _might_ (I say might because I don't have to dive into these kinds of use cases in my daily work) have to go with JVM GC major collections and rendering... Here are some screenshots of the Arch task manager and Chrome Devtools from beta 1... 

I'm having a hard time believing this is a Ferdium issue, but I could be wrong. On Chromium, using the web app, the same processes are at play but seem to be less of an issue.

TL;DR - Garbage collection is CPU intensive and it looks like Proton is doing quite large cleanups.

![image](https://user-images.githubusercontent.com/55257671/172757490-0bb7e59e-5ae2-402d-85b5-bb8e088dfae0.png)
![Screenshot_2022-06-08_23-22-47](https://user-images.githubusercontent.com/55257671/172757521-4726934c-facc-4a35-ac6e-3deda0b3ec3b.png)

@vraravam do you think my assumption here is correct?

**Comment by pepper3k** [comment]
I noticed that this seems to have been resolved from version 56 and up. I am currently on 58 and have about 12 services open, and my CPU is averaging at about 1%, none of the ferdium services appear at the top of the list, they appear to be at 0%. Once in a while one of them will "spike" up to about 1 or 2 percent for a moment and then go back down to 0.

I have a feeling that this was resolved on proton's infrastructure, or coding in their webapp backend.

 @palepinkdot are you still experiencing this issue?

@vraravam 
If no one is having this problem anymore, I think it would be safe to close this one off...

**Comment by vraravam** [comment]
Thanks for the updates @eitanprezes - closing this now!
